editor/plugins/control_editor_plugin.cpp

```diff
@@ -448,37 +450,266 @@ bool EditorInspectorPluginControl::parse_property(Object *p_object, const Varian
+			preset_buttons[PRESET_RIGHT_WIDE]->set_icon(get_theme_icon(SNAME("ControlAlignRightWide"), SNAME("EditorIcons")));
+
+			preset_buttons[PRESET_FULL_RECT]->set_icon(get_theme_icon(SNAME("ControlAlignFullRect"), SNAME("EditorIcons")));
+		} break;
+	}
+}
+
+void ControlAnchorPresetPicker::_bind_methods() {
+	ADD_SIGNAL(MethodInfo("preset_selected", PropertyInfo(Variant::INT, "preset")));
+}
+
+ControlAnchorPresetPicker::ControlAnchorPresetPicker() {
+	VBoxContainer *main_vb = memnew(VBoxContainer);
+	main_vb->add_theme_constant_override("separation", grid_separation);
+	add_child(main_vb);
+
+	HBoxContainer *top_row = memnew(HBoxContainer);
+	top_row->set_alignment(BoxContainer::ALIGNMENT_CENTER);
+	top_row->add_theme_constant_override("separation", grid_separation);
+	main_vb->add_child(top_row);
+
+	_add_row_button(top_row, PRESET_TOP_LEFT, TTR("Top Left"));
+	_add_row_button(top_row, PRESET_CENTER_TOP, TTR("Center Top"));
+	_add_row_button(top_row, PRESET_TOP_RIGHT, TTR("Top Right"));
+	_add_separator(top_row, memnew(VSeparator));
+	_add_row_button(top_row, PRESET_TOP_WIDE, TTR("Top Wide"));
+
+	HBoxContainer *mid_row = memnew(HBoxContainer);
+	mid_row->set_alignment(BoxContainer::ALIGNMENT_CENTER);
+	mid_row->add_theme_constant_override("separation", grid_separation);
+	main_vb->add_child(mid_row);
+
+	_add_row_button(mid_row, PRESET_CENTER_LEFT, TTR("Center Left"));
+	_add_row_button(mid_row, PRESET_CENTER, TTR("Center"));
+	_add_row_button(mid_row, PRESET_CENTER_RIGHT, TTR("Center Right"));
+	_add_separator(mid_row, memnew(VSeparator));
+	_add_row_button(mid_row, PRESET_HCENTER_WIDE, TTR("HCenter Wide"));
+
+	HBoxContainer *bot_row = memnew(HBoxContainer);
+	bot_row->set_alignment(BoxContainer::ALIGNMENT_CENTER);
+	bot_row->add_theme_constant_override("separation", grid_separation);
+	main_vb->add_child(bot_row);
+
+	_add_row_button(bot_row, PRESET_BOTTOM_LEFT, TTR("Bottom Left"));
+	_add_row_button(bot_row, PRESET_CENTER_BOTTOM, TTR("Center Bottom"));
+	_add_row_button(bot_row, PRESET_BOTTOM_RIGHT, TTR("Bottom Right"));
+	_add_separator(bot_row, memnew(VSeparator));
+	_add_row_button(bot_row, PRESET_BOTTOM_WIDE, TTR("Bottom Wide"));
+
+	_add_separator(main_vb, memnew(HSeparator));
+
+	HBoxContainer *extra_row = memnew(HBoxContainer);
+	extra_row->set_alignment(BoxContainer::ALIGNMENT_CENTER);
+	extra_row->add_theme_constant_override("separation", grid_separation);
+	main_vb->add_child(extra_row);
+
+	_add_row_button(extra_row, PRESET_LEFT_WIDE, TTR("Left Wide"));
+	_add_row_button(extra_row, PRESET_VCENTER_WIDE, TTR("VCenter Wide"));
+	_add_row_button(extra_row, PRESET_RIGHT_WIDE, TTR("Right Wide"));
+	_add_separator(extra_row, memnew(VSeparator));
+	_add_row_button(extra_row, PRESET_FULL_RECT, TTR("Full Rect"));
+}
+
+void ContainerSizeFlagPicker::_add_row_button(HBoxContainer *p_row, const SizeFlags p_flag, const String &p_name) {
```

Comment:
This duplicates ControlAnchorPresetPicker. I wonder if it's possible to merge these two into one class 🤔 (you could replace constructors with initialization methods)

Reply:
Could make it into a helper method outside of the class. We don't use that pattern often, but sometimes we do.

Reply:
No I commented here, but the classes have really much in common, including a constant that is defined twice (`grid_separation`). So maybe it's possible to merge them into one class and use setup methods for initialization instead of constructors. Much of their code wold be shared. I think the only problem would be methods that take flags as argument. You'd probably need to make them into ints.